server/src/main/java/org/opensearch/index/codec/composite/LuceneDocValuesConsumerFactory.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.composite;
+
+import org.apache.lucene.codecs.DocValuesConsumer;
+import org.apache.lucene.codecs.lucene90.Lucene90DocValuesConsumerWrapper;
+import org.apache.lucene.index.SegmentWriteState;
+
+import java.io.IOException;
+
+/**
+ * A factory class that provides a factory method for creating {@link DocValuesConsumer} instances
+ * based on the specified composite codec.
+ *
+ * @opensearch.experimental
+ */
+public class LuceneDocValuesConsumerFactory {
+
+    public static DocValuesConsumer getDocValuesConsumerForCompositeCodec(
```

Comment:
Add more documentation around why we are directly using `Lucene90DocValuesConsumerWrapper` vs why we need to check `Composite99Codec.COMPOSITE_INDEX_CODEC_NAME` and then use Lucene90DocValuesProducerWrapper.

Reply:
+1, don't we need `compositeCodec` parameter here?

Reply:
we don't need the composite codec because we will always write with the latest composite codec. The codec will make sure to call the consumer factory for its abstractions.
In producers, we had to have a switch case to make sure we are able to read using older codecs as well.

In a nutshell, we write using the latest codec only and ensuring read through the all present codecs in the segment

Reply:
I will add the documentation too.